Slider/Assets/Scripts/UI/Artifact/Tabs/DesertTabManager.cs

```diff
@@ -43,53 +58,180 @@ public override void SetCurrentScreen(int screenIndex)
+        }
     }
 
-    private void UpdateMiddleEmpty()
+    private void UpdateFragSwapTile()
     {
-        middle = UIArtifact.GetButton(1, 1);
-        empty = uiArtifactMenus.uiArtifact.GetButton(9);
+        if (fragSwapTileCycleRoutine != null)
+        {
+            StopCoroutine(fragSwapTileCycleRoutine);
+        }
+
+        ArtifactTileButton tile8 = uiArtifactMenus.uiArtifact.GetButton(8);
+        //do we have a better way to check if there's more than one empty tile?
+        if (!tile8.TileIsActive)
+        {
+            fragSwapTile = tile8;
+            fragSwapTileCycleRoutine = StartCoroutine(CycleFragSwapTileBetweenEmptyTiles());
+        }
+        else
+        {
+            fragSwapTile = uiArtifactMenus.uiArtifact.GetButton(9);
+            SetFragSwapTile(fragSwapTile);
+        }
+    }
+
+    private IEnumerator CycleFragSwapTileBetweenEmptyTiles()
+    {
+        ArtifactTileButton[] emptyTiles = { uiArtifactMenus.uiArtifact.GetButton(8), uiArtifactMenus.uiArtifact.GetButton(9) };
+        
+        while (true)
+        {
+            foreach(ArtifactTileButton emptyTile in emptyTiles)
+            {
+                fragSwapTile = emptyTile;
+                SetFragSwapTile(fragSwapTile);
+                yield return new WaitForSeconds(1);
+                fragSwapTile.SetScrollHighlight(false);
+            }
+        }
+    }
+
+    private void SetFragSwapTile(ArtifactTileButton tile)
+    {
+        tile.SetScrollHighlight(true);
+
+        switch (tile.x)
+        {
+            case (0):
+                switch (tile.y)
+                {
+                    case (0):
+                        rearrangingFragTabImage.sprite = tabSprites[5];
+                        break;
+                    case (1):
+                        rearrangingFragTabImage.sprite = tabSprites[3];
+                        break;
+                    case (2):
+                        rearrangingFragTabImage.sprite = tabSprites[0];
+                        break;
+                }
+                break;
+            case (1):
+                switch (tile.y)
+                {
+                    case (0):
+                        rearrangingFragTabImage.sprite = tabSprites[6];
+                        break;
+                    case (2):
+                        rearrangingFragTabImage.sprite = tabSprites[1];
+                        break;
+                }
+                break;
+            case (2):
+                switch (tile.y)
+                {
+                    case (0):
+                        rearrangingFragTabImage.sprite = tabSprites[7];
+                        break;
+                    case (1):
+                        rearrangingFragTabImage.sprite = tabSprites[4];
+                        break;
+                    case (2):
+                        rearrangingFragTabImage.sprite = tabSprites[2];
+                        break;
+                }
+                break;
+        }
```

Comment:
can u just make a method to build/cache a 2D array and access it with like `rearrangingTabSprites[x][y]`